geode-core/src/integrationTest/java/org/apache/geode/net/SSLSocketParameterExtensionIntegrationTest.java

```diff
@@ -0,0 +1,242 @@
+  private SocketCreator socketCreator;
+  private InetAddress localHost;
+  private Thread serverThread;
+  private ServerSocket serverSocket;
+  private Socket clientSocket;
+  private InternalCache cache;
+
+  @Rule
+  public ErrorCollector errorCollector = new ErrorCollector();
+
+  @Rule
+  public RestoreSystemProperties restoreSystemProperties = new RestoreSystemProperties();
+
+  @Rule
+  public TemporaryFolder temporaryFolder = new TemporaryFolder();
+
+  @Rule
+  public TestName testName = new TestName();
+
+
+  private Throwable serverException;
+
+  @Before
+  public void setUp() throws Exception {
+    IgnoredException.addIgnoredException("javax.net.ssl.SSLException: Read timed out");
+
+    File keystore = findTestKeystore();
+    System.setProperty("javax.net.ssl.trustStore", keystore.getCanonicalPath());
+    System.setProperty("javax.net.ssl.trustStorePassword", "password");
+    System.setProperty("javax.net.ssl.keyStore", keystore.getCanonicalPath());
+    System.setProperty("javax.net.ssl.keyStorePassword", "password");
+
+    Properties properties = new Properties();
+    properties.setProperty(MCAST_PORT, "0");
+    properties.setProperty(CLUSTER_SSL_ENABLED, "true");
+    properties.setProperty(CLUSTER_SSL_REQUIRE_AUTHENTICATION, "true");
+    properties.setProperty(CLUSTER_SSL_CIPHERS, "any");
+    properties.setProperty(CLUSTER_SSL_PROTOCOLS, "TLSv1.2");
+    properties.setProperty(SSL_PARAMETER_EXTENSION, MySSLParameterExtension.class.getName());
+    properties.setProperty(DISTRIBUTED_SYSTEM_ID, "11");
+
+    cache = (InternalCache) new CacheFactory(properties).create();
+
+    this.distributionConfig = new DistributionConfigImpl(properties);
+
+    SocketCreatorFactory.setDistributionConfig(this.distributionConfig);
+    this.socketCreator = SocketCreatorFactory.getSocketCreatorForComponent(CLUSTER);
+
+    this.localHost = InetAddress.getLocalHost();
+  }
+
+  @After
+  public void tearDown() throws Exception {
+    if (this.clientSocket != null) {
+      this.clientSocket.close();
+    }
+    if (this.serverSocket != null) {
+      this.serverSocket.close();
+    }
+    if (this.serverThread != null && this.serverThread.isAlive()) {
+      this.serverThread.interrupt();
+    }
+    SocketCreatorFactory.close();
+    cache.close();
+  }
+
+  @Test
+  public void securedSocketCheckExtensions() throws Exception {
+    this.serverSocket = this.socketCreator.createServerSocket(0, 0, this.localHost);
+    this.serverThread = startServer(this.serverSocket, 15000);
+
+    int serverPort = this.serverSocket.getLocalPort();
+    this.clientSocket = this.socketCreator.connectForServer(this.localHost, serverPort);
+
+    SSLSocket sslSocket = (SSLSocket) this.clientSocket;
+
+    List<SNIServerName> serverNames = new ArrayList<>(1);
+    SNIHostName serverName = new SNIHostName("11");
+    serverNames.add(serverName);
+
+    assertEquals(sslSocket.getSSLParameters().getServerNames(), serverNames);
```

Comment:
Just a minor nit, but I think most new tests are using the AssertJ style for assertions. In this case that would be:
```
    assertThat(sslSocket.getSSLParameters().getServerNames()).isEqualTo(serverNames);
```